app_dart/lib/src/service/datastore.dart

```diff
@@ -151,7 +151,8 @@ class DatastoreService {
         updates: 0,
       );
     } else {
-      assert(previousStatusUpdates.length == 1);
+      assert(previousStatusUpdates.length == 1,
```

Comment:
We're not running with asserts on GCP.

Reply:
```suggestion
      assert(previousStatusUpdates.length == 1);

      if (previousStatusUpdates.length != 1) {
        print('GithubBuildStatusUpdate should have no more than one entries on repository $slug.fullName, pr $pr.number, head $pr.head.sha');
      }

      return previousStatusUpdates[0];
```
If this is an urgent issue, we could just start by printing to stdout. I believe GCP is capturing stdout of the server file and logging it. Otherwise, we should probably inject Logger to this class and start logging this specific case. Would be good to find out what caused this to create duplicate keys.

Reply:
Changed to throwing an exception instead. How about this way?

Reply:
SGTM. I didn't know if we wanted it to make a best effort approach.

Reply:
Having multiple entries in here with the same hash indicates something has gone wrong or our understanding of the way all this is working is wrong.

Although maybe something got mixed up where multiple PRs had the same hashes in them for some reason.  Perhaps because of something like https://github.com/flutter/flutter/pull/48985